internal/services/storage/storage_account_queue_properties_resource.go

```diff
@@ -0,0 +1,666 @@
+		},
+		"logging": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Computed: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"version": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"delete": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
+					},
+					"read": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
+					},
+					"write": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
+					},
+					"retention_policy_days": {
+						Type:         pluginsdk.TypeInt,
+						Optional:     true,
+						ValidateFunc: validation.IntBetween(1, 365),
+					},
+				},
+			},
+			AtLeastOneOf: []string{"minute_metrics", "hour_metrics", "logging", "cors_rule"},
+		},
+		"minute_metrics": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Computed: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"version": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"include_apis": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  false,
+					},
+					"retention_policy_days": {
+						Type:         pluginsdk.TypeInt,
+						Optional:     true,
+						ValidateFunc: validation.IntBetween(1, 365),
+					},
+				},
+			},
+			AtLeastOneOf: []string{"minute_metrics", "hour_metrics", "logging", "cors_rule"},
+		},
+	}
+}
+
+func (s AccountQueuePropertiesResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (s AccountQueuePropertiesResource) ModelObject() interface{} {
+	return &AccountQueuePropertiesModel{}
+}
+
+func (s AccountQueuePropertiesResource) ResourceType() string {
+	return "azurerm_storage_account_queue_properties"
+}
+
+func (s AccountQueuePropertiesResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return commonids.ValidateStorageAccountID
+}
+
+func (s AccountQueuePropertiesResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			storageClient := metadata.Client.Storage
+			var model AccountQueuePropertiesModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			accountID, err := commonids.ParseStorageAccountID(model.StorageAccountId)
+			if err != nil {
+				return err
+			}
+
+			// Get the target account to ensure it supports queues
+			account, err := storageClient.ResourceManager.StorageAccounts.GetProperties(ctx, *accountID, storageaccounts.DefaultGetPropertiesOperationOptions())
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *accountID, err)
+			}
+			if account.Model == nil {
+				return fmt.Errorf("account %s has no model", *accountID)
+			}
```

Comment:
We've been returning these as 
```suggestion
			if account.Model == nil {
				return fmt.Errorf("retrieving %s: `model` was nil", *accountID)
			}
```